src/NLogTarget/ApplicationInsightsTarget.cs

```diff
@@ -127,7 +130,17 @@ protected override void FlushAsync(AsyncContinuation asyncContinuation)
             try
             {
                 this.TelemetryClient.Flush();
-                asyncContinuation(null);
+                if (DateTime.UtcNow.AddSeconds(-30) > this.lastLogEventTime)
+                {
+                    // Nothing has been written, so nothing to wait for
+                    asyncContinuation(null);
+                }
+                else
+                {
+                    // Documentation says it is important to wait after flush, else nothing will happen
+                    // https://docs.microsoft.com/en-us/azure/application-insights/app-insights-api-custom-events-metrics#flushing-data
```

Comment:
please remove `en-US` from link so it will direct to localized version of the doc

Reply:
Fixed